apiserver/facades/controller/agenttools/agenttools.go

```diff
@@ -94,21 +98,21 @@ func checkToolsAvailability(ctx context.Context, newEnviron newEnvironFunc, mode
 	return newest, nil
 }
 
-var modelConfig = func(e *state.Model) (*config.Config, error) {
-	return e.Config()
-}
-
 // Base implementation of envVersionUpdater
 func envVersionUpdate(env *state.Model, ver version.Number) error {
 	return env.UpdateLatestToolsVersion(ver)
 }
 
-func updateToolsAvailability(ctx context.Context, modelGetter ModelGetter, newEnviron newEnvironFunc, finder toolsFinder, update envVersionUpdater, logger corelogger.Logger) error {
-	model, err := modelGetter.Model()
-	if err != nil {
-		return errors.Annotate(err, "cannot get model")
-	}
-	cfg, err := modelConfig(model)
+func updateToolsAvailability(
```

Comment:
We are going to need to make some changes here.

Specifically using model config to get the models agent version is something we are actively stamping out.

At the moment we have been replicating the agent version set on the agent version table in the controller database into the read only model information. This was a bad design decision that we have to undo now.

What I propose as part of this PR is that we introduce a new service for interacting with a models agent information. We can nut out the design together and quickly get it implemented.

Reply:
I am also thinking that we can avoid passing all the services into these functions and just load them up with the information they need.

Reply:
Honestly this whole facade probably needs to be redone, it's using some bad design patterns, but I don't know if we should do that in this PR.

Reply:
P.S. are you talking about pulling some of this facade logic into the new service?